src/Controls/src/Build.Tasks/CompiledMarkupExtensions/StaticResourceExtension.cs

```diff
@@ -0,0 +1,179 @@
+            var localName = propertyName.LocalName;
+            var parentType = module.ImportReference((node.Parent as IElementNode).XmlType.GetTypeReference(context.Cache, module, (IXmlLineInfo)node));
+
+            var bpRef = SetPropertiesVisitor.GetBindablePropertyReference(parentType, propertyName.NamespaceURI, ref localName, out _, context, (IXmlLineInfo)node);
+            //BindableProperty
+            if (bpRef != null) 
+            {
+                var targetTypeRef = module.ImportReference(bpRef.GetBindablePropertyType(context.Cache, node as IXmlLineInfo, module));
+                foreach (var instruction in stringResourceNode.PushConvertedValue(context, bpRef, stringResourceNode.PushServiceProvider(context, bpRef: bpRef), true, false))
+					yield return instruction;
+                var vardef = new VariableDefinition(targetTypeRef);
+                yield return Create(Stloc, vardef);
+                vardefref.VariableDefinition = vardef;
+                yield break;
+            }
+            
+            var propertyRef = parentType.GetProperty(context.Cache, pd => pd.Name == localName, out var declaringTypeReference);
+            if (propertyRef != null)
+            {
+                foreach (var instruction in stringResourceNode.PushConvertedValue(context, propertyRef.PropertyType, new ICustomAttributeProvider[] { propertyRef, propertyRef.PropertyType.ResolveCached(context.Cache) }, stringResourceNode.PushServiceProvider(context, propertyRef: propertyRef), true, false))
+					yield return instruction;
+                var vardef = new VariableDefinition(propertyRef.PropertyType);
+                yield return Create(Stloc, vardef);
+                vardefref.VariableDefinition = vardef;
+                yield break;
+            }
+
+
+        }
+
+        public static IEnumerable<Instruction> FallBack(string key, IElementNode node, ModuleDefinition module, ILContext context)
+        {
+            var staticResourceExtensionType = module.ImportReference(context.Cache, 
+                                                                    ("Microsoft.Maui.Controls.Xaml", "Microsoft.Maui.Controls.Xaml", "StaticResourceExtension"));
+            yield return Create(Newobj, module.ImportCtorReference(context.Cache, 
+                                                                    staticResourceExtensionType, 
+                                                                    paramCount: 0));
+            
+            SetPropertiesVisitor.TryGetPropertyName(node, node.Parent, out var propertyName);
+            var localName = propertyName.LocalName;
+
+            //Set the Key
+			var keyProperty = staticResourceExtensionType.GetProperty(context.Cache, pd => pd.Name == "Key", out _);
+            yield return Create(Dup);
+            yield return Create(Ldstr, key);
+            yield return Create(Callvirt, module.ImportReference(keyProperty.SetMethod));
```

Comment:
This is creating a duplicate `StaticResourceExtension` instance, this is what I see in decompiled code:
```c#
staticResourceExtension.Key = "Primary";
StaticResourceExtension obj = new StaticResourceExtension
{
    Key = "Primary"
};
```
Could we reuse the original `StaticResourceExtension` instance instead of creating a new one?